pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,306 @@
+            Self::execute_call_(origin, &call)?;
+            Ok(Pays::No.into())
+        }
+
+        /// Set the Update Frequency through Root
+        #[weight = 10_000 + T::DbWeight::get().writes(1)]
+        pub fn root_set_update_freq(origin, new_update_freq: T::BlockNumber) -> DispatchResult {
+            ensure_root(origin)?;
+            Self::set_update_freq_(new_update_freq);
+            Ok(())
+        }
+
+        /// Set DockFiatRate through Root
+        #[weight = 10_000 + T::DbWeight::get().writes(1)]
+        pub fn root_set_dock_fiat_rate(origin, new_rate: Permill) -> DispatchResult {
+            ensure_root(origin)?;
+            Self::set_dock_fiat_rate_(new_rate);
+            Ok(())
+        }
+    }
+}
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+
+type AmountUsd = Permill;
+
+// private helper functions
+impl<T: Config> Module<T> {
+    fn get_call_fee_fiat_(
+        call: &<T as Config>::Call,
+    ) -> Result<AmountUsd, DispatchErrorWithPostInfo> {
+        match call.is_sub_type() {
+            Some(did::Call::new(_did, _detail)) => return Ok(Permill::from_percent(50)),
+            Some(did::Call::update_key(_key_update, _sig)) => return Ok(Permill::from_percent(51)),
+            Some(did::Call::remove(_to_remove, _sig)) => return Ok(Permill::from_percent(52)),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(anchor::Call::deploy(_bytes)) => return Ok(Permill::from_percent(60)),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(blob::Call::new(_blob, _sig)) => return Ok(Permill::from_percent(70)),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(revoke::Call::new_registry(_id, _registry)) => {
+                return Ok(Permill::from_percent(80))
+            }
+            Some(revoke::Call::revoke(_revoke, _proof)) => return Ok(Permill::from_percent(81)),
+            Some(revoke::Call::unrevoke(_unrevoke, _proof)) => {
+                return Ok(Permill::from_percent(82))
+            }
+            Some(revoke::Call::remove_registry(_rm, _proof)) => {
+                return Ok(Permill::from_percent(83))
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(attest::Call::set_claim(_attester, _att, _sig)) => {
+                return Ok(Permill::from_percent(40))
+            }
+            _ => {}
+        }
+
+        // return error if Call not in list
+        // uses Pays::Yes to prevent spam
+        return Err(Error::<T>::UnexpectedCall.into());
+    }
+    fn compute_call_fee_dock_(
+        call: &<T as Config>::Call,
+    ) -> Result<BalanceOf<T>, DispatchErrorWithPostInfo> {
+        use sp_std::convert::TryInto;
+        let fee_fiat_permill: AmountUsd = Self::get_call_fee_fiat_(call)?;
```

Comment:
Wouldn't it be simpler and simpler and efficient to not use Permill and instead use u64 (our Balance type) and do (Self::get_call_fee_fiat_(call)?.saturating_mul(1_000_000).checked_div(Self::dock_fiat_rate()))